### PR TITLE
Add example to std::process::abort

### PR DESCRIPTION
**Pull request by rap2hpoutre**
This is a second step in order to complete this issue: https://github.com/rust-lang/rust/issues/29370
I submitted this PR with the help of @steveklabnik again. Thanks to him! More info here: https://github.com/rust-lang/rust/issues/29370#issuecomment-290653877

### CODE REVIEW

**Review by steveklabnik — APPROVED** [review]

File: src/libstd/process.rs [hunk 1]
```diff
@@ -1070,6 +1070,28 @@ pub fn exit(code: i32) -> ! {
...
+///     }
+/// }
+///
+/// fn main() {
+///     let _x = HasDrop;
+///     process::abort();
+///     // the destructor implemented for HasDrop will never get run
+/// }
+/// ```
+///
```
Comment (steveklabnik) [inline src/libstd/process.rs]: could you remove this line, please?

**Review by steveklabnik — CHANGES_REQUESTED** [review]
This is fantastic! One tiny, tiny thing though, then this is good to go

File: src/libstd/process.rs [hunk 2]
```diff
@@ -1070,6 +1070,28 @@ pub fn exit(code: i32) -> ! {
 ///     // execution never gets here
 /// }
 /// ```
+///
+/// The abort function terminates the process, so the destructor will not get
```
Comment (steveklabnik) [inline src/libstd/process.rs]: oh and could this link to abort please?

**Review by steveklabnik — CHANGES_REQUESTED** [review]

File: src/libstd/process.rs (hunk 2 above)
Comment (frewsxcv) [inline src/libstd/process.rs]: There's also whitespace at the end of this line the tidy checker doesn't like.

File: src/libstd/process.rs (hunk 1 above)
Comment (rap2hpoutre) [inline src/libstd/process.rs]: Ok done!

File: src/libstd/process.rs (hunk 2 above)
Comment (rap2hpoutre) [inline src/libstd/process.rs]: Ok for both comments! Thanks!

**Comment by rap2hpoutre** [comment]
@steveklabnik Ok, everything seems fine now. Thank you again!

**Review by steveklabnik — APPROVED** [review]

**Comment by steveklabnik** [comment]
@bors: r+ rollup

💯 ❤️ ✨ thank you!!!

**Pull request merged**